File: src-scala/org/jgll_staged/grammar/condition/ContextFreeCondition.scala
package org.jgll_staged.grammar.condition

import java.util.List
import org.jgll_staged.grammar.Symbol
import org.jgll_staged.util.CollectionsUtil._
import scala.reflect.{BeanProperty, BooleanBeanProperty}
//remove if not needed
import scala.collection.JavaConversions._

@SerialVersionUID(1L)
class ContextFreeCondition(`type`: ConditionType, @BeanProperty var symbols: List[_ <: Symbol])
    extends Condition(`type`) {

  override def toString(): String = {
    `type`.toString + " " + listToString(symbols)
  }
}

File: java/src/org/jgll/grammar/condition/ContextFreeCondition.java
package org.jgll.grammar.condition;

import java.util.List;

import org.jgll.grammar.Symbol;

import static org.jgll.util.CollectionsUtil.*;

public class ContextFreeCondition extends Condition {
	
	private static final long serialVersionUID = 1L;
	
	private List<? extends Symbol> symbols;

	public ContextFreeCondition(ConditionType type, List<? extends Symbol> symbols) {
		super(type);
		this.symbols = symbols;
	}
	
	public List<? extends Symbol> getSymbols() {
		return symbols;
	}
	
	@Override
	public String toString() {
		return type.toString() + " " + listToString(symbols);
	}
}
